Src/Base/AMReX_MPMD.H

```diff
@@ -26,18 +26,27 @@ int MyProgId (); //! Program ID
 class Copier
 {
 public:
-    Copier (BoxArray const& ba, DistributionMapping const& dm);
+    Copier ();
+
+    Copier (BoxArray const& ba, DistributionMapping const& dm,
+            bool send_ba = false);
 
     template <typename FAB>
     void send (FabArray<FAB> const& mf, int icomp, int ncomp) const;
 
     template <typename FAB>
     void recv (FabArray<FAB>& mf, int icomp, int ncomp) const;
 
+    BoxArray get_my_ba();
+
+    DistributionMapping get_my_dm();
+
 private:
     std::map<int,FabArrayBase::CopyComTagsContainer> m_SndTags;
     std::map<int,FabArrayBase::CopyComTagsContainer> m_RcvTags;
     bool m_is_thread_safe;
+    Vector<Box> m_bv;
+    DistributionMapping m_dm;
```

Comment:
```suggestion
    BoxArray m_ba;
    DistributionMapping m_dm;
```
BoxArray has a shared_ptr inside. If we store Vector<Box> instead of BoxArray, everytime `get_my_ba` is called a completely new BoxArray without sharing is created. We should try to avoid that.